### DISCUSSION TITLE
Dynamic public files/images after build

### DISCUSSION THREAD

**Discussion by Megarsh** [nuxt/nuxt#17144] [Questions]
Am I correct to assume that I can't serve dynamically added public files/images via nuxt after build?

Everything works fine in dev mode, but for production I can't figure out how to add new images because everything that's in public directory is assigned to an assets array (inside static.mjs) during build.

**Comment by Megarsh** [comment]
Was able to solve this by changing my image directory and adding a server middleware + something similar to a serveStatic method
Though I'm not sure if this is a correct/good approach

    serverMiddleware: [
        { path: '/content', handler: './static' }
    ]

```
import { promises } from 'fs'
import config from '#config'

const TWO_DAYS = 2 * 60 * 60 * 24
//TODO add more types and file check
const types = {
    png: 'image/png',
    jpg: 'image/jpeg',
    svg: 'image/svg+xml'
}

export default async function (req, res) {
    try {
        if (req.headers['if-modified-since']) {
            res.statusCode = 304
            return res.end('Not Modified')
        } else {
            const content = await promises.readFile(config.staticPath + req.url)

            res.setHeader('Cache-Control', `max-age=${TWO_DAYS}, immutable`)
            res.setHeader('Content-Type', types[req.url.split('.').pop()])
            res.setHeader('Last-Modified', (new Date()).toISOString())

            return res.end(content)
        }
    } catch (err) {
        return res.statusCode = 404
    }
}
```

**Comment by itshizhan** [comment]
we did this by nginx

**Reply by Megarsh** [reply]
Ended up doing exactly this, I've completely removed the public folder and decided to handle all images outside of nuxt which feels really unintuitive.

**Reply by Waltem** [reply]
> all images outside of nuxt

put all files outside .output folder? how u read these files?
Please, can u explain how to use nginx? Docker?
